packages/core/src/auth/auth.ts

```diff
@@ -506,6 +508,12 @@ export class Auth implements AuthService, ConnectionManager {
 
     private async updateConnectionState(id: Connection['id'], connectionState: ProfileMetadata['connectionState']) {
         getLogger().info(`auth: Updating connection state of ${id} to ${connectionState}`)
+
+        if (connectionState === 'authenticating') {
+            this.#onDidChangeConnectionState.fire({ id, state: connectionState })
+            return
+        }
```

Comment:
This seems like a special case but perhaps it's harmless.

Reply:
This looks like we are changing behavior. Now if the state is set as `authenticating` we will not store it in the metadata using `this.store.updateMetadata()` below.

What is the reason?